src/boot.lisp

```diff
@@ -93,13 +93,40 @@
      ,@(when (stringp docstring) `((oset ,docstring ',name "vardoc")))
      ',name))
 
-(defmacro defun (name args &rest body)
+;;; (defun (setf))
+(defmacro %defun (name args &rest body)
   `(progn
      (eval-when (:compile-toplevel)
        (fn-info ',name :defined t))
      (fset ',name #'(named-lambda ,name ,args ,@body))
      ',name))
 
+(defmacro defun (name args &rest body)
+    (cond ((symbolp name)
+           `(%defun ,name ,args ,@body))
+          ((and (consp name) (eq (car name) 'setf))
+           (let ((sfn 
```

Comment:
We shouldn't intern symbols like this that the user is not expecting.

Do we have to? should we use an uninterned symbol, or even better, should we store the list of `(setf F)` function specifiers in a hash table so they can be found by `fdefinition`?

Reply:
I'm aware that using a hash table at that early point (`boot.lisp`) is probably not an option. Should  we can patch `defun` a bit later on?  But better ideas are welcome.

If it is still tricky, we can go with the uninterned symbol solution for now.

Reply:
David, thanks for the review.

I know that this is not an ideal solution.
But, I would not like to make changes to several places at once (defun/setf-expander/setf).
This minimal solution in the future can be replaced without any problems.

As for the user's expectations:
- If the user uses the form `(defun (setf F) ...)` he is, higly likely, not interested fbounded symbols, and it does not see `*setf-expanders*`.
- If the user uses `define-setf-expander` then he himself controls the names.

Use `hash-table` in `setf-expander` this is a good idea. But it requires a lot of time to implement.
I'm not sure that I have such a time now.